sdk/core/core/CMakeLists.txt

```
@@ -4,8 +4,41 @@
 cmake_minimum_required (VERSION 3.10)
 set(TARGET_NAME "az_core")
 
-project (${TARGET_NAME} LANGUAGES C)
+# Read in az_version.h
+#   The build will fail with an error if the file is not found or improperly formatted
+set(AZ_SDK_VERSION_H_FILE "${CMAKE_CURRENT_SOURCE_DIR}/inc/az_version.h")
 
+if(EXISTS ${AZ_SDK_VERSION_H_FILE})
+    file (STRINGS ${AZ_SDK_VERSION_H_FILE} VERSION_H_CONTENT)
+    message(STATUS "Read version from ${AZ_SDK_VERSION_H_FILE}")
+
+    #Find and read the AZ_SDK_VERSION_STRING define
+    string(REGEX MATCHALL "#define AZ_SDK_VERSION_STRING \"([0-9]+)\.([0-9]+)\.([0.9]+)([^\"]*)\"" VERSION_PARTS ${VERSION_H_CONTENT})
```

Is `0.9` expected, or is that a typo and should be 0-9?
```suggestion
    string(REGEX MATCHALL "#define AZ_SDK_VERSION_STRING \"([0-9]+)\.([0-9]+)\.([0-9]+)([^\"]*)\"" VERSION_PARTS ${VERSION_H_CONTENT})
```

---

Yep, fixed as part of the prerelease hardened checks.